### PR TITLE
Refactor relnotes tool

### PR DESCRIPTION
**Pull request by hoegaarden**
As a followup to #973, another round of smaller refactors and cleanups.

/kind cleanup
/priority P3
/assign @saschagrunert 
/milestone v1.18

### CODE REVIEW

**Comment by hoegaarden** [comment]
/retest

@saschagrunert  -- maybe [a race](https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/release/977/pull-release-test/1206569367219736576#1:build-log.txt%3A92)?

**Comment by saschagrunert** [comment]
> /retest
> 
> @saschagrunert -- maybe [a race](https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/release/977/pull-release-test/1206569367219736576#1:build-log.txt%3A92)?

I think that's a race, yes.

File: cmd/release-notes/README.md [hunk 1]
```diff
@@ -75,7 +75,7 @@ level=debug timestamp=2019-07-30T04:02:44.3716249Z caller=notes.go:497 msg="Excl
 | github-token | GITHUB_TOKEN | | Yes | A personal GitHub access token |
 | github-org | GITHUB_ORG | kubernetes | Yes | Name of GitHub organization |
 | github-repo | GITHUB_REPO | kubernetes | Yes | Name of GitHub repository |
-| requiredAuthor | REQUIRED_AUTHOR | k8s-ci-robot | Yes | Only commits from this GitHub user are considered. Set to empty string to include all users |
+| required-author | REQUIRED_AUTHOR | k8s-ci-robot | Yes | Only commits from this GitHub user are considered. Set to empty string to include all users |
```
Comment (saschagrunert) [inline cmd/release-notes/README.md]: This seems a breaking change but I think it might be okay (last famous words).

File: pkg/notes/notes.go [hunk 2]
```diff
@@ -399,32 +357,28 @@ func ReleaseNoteFromCommit(result *Result, relVer string, opts ...GitHubAPIOptio
...
 		}
-		return resp, err
+		return commits, resp, err
 	}
 
 	clo := github.CommitsListOptions{
-		SHA:   c.branch,
+		SHA:   branch,
 		Since: *startCommit.Committer.Date,
 		Until: *endCommit.Committer.Date,
```
Comment (saschagrunert) [inline pkg/notes/notes.go]: I assume this can never be `nil` so it's fine, right?

**Comment by saschagrunert** [comment]
/hold 

for more eyes on it

File: cmd/release-notes/README.md (hunk 1 above)
Comment (hoegaarden) [reply cmd/release-notes/README.md]: I found, that the flags have a deprecation method and use that now. See https://github.com/kubernetes/release/pull/977/files#diff-91de0979ac440515ffbed7847c8691aeR190-R194.

File: pkg/notes/notes.go (hunk 2 above)
Comment (hoegaarden) [reply pkg/notes/notes.go]: I certainly could be `nil`. Those structs are built with expressing everything as pointers. Also it has been like that all along.
However, the structs also have methods to access that data, so I switched to using that.

File: cmd/release-notes/main.go [hunk 3]
```diff
@@ -262,6 +268,13 @@ func GetReleaseNotes() (notes.ReleaseNotes, notes.ReleaseNotesHistory, error) {
 	return releaseNotes, history, nil
 }
 
+func or(opt, def string) string {
```
Comment (justaugustus) [inline cmd/release-notes/main.go]:
```suggestion
func or(opt, default string) string {
```

Let's add a function comment here.

**Comment by hoegaarden** [comment]
/hold
until #984 is managed/resolved.

File: cmd/release-notes/main.go (hunk 3 above)
Comment (hoegaarden) [reply cmd/release-notes/main.go]: `default` is a reserved word: https://golang.org/ref/spec#Keywords.
However, that func is actually not needed, as we already do defaulting on the `opts` struct.

**Pull request merged**